pythoncode/FortiusAntBody.py

```diff
@@ -462,125 +485,130 @@ def Tacx2Dongle(self):
+            #-------------------------------------------------------------------
             msg = ant.msg4D_RequestMessage(ant.channel_HRM_s, ant.msgID_ChannelID)
-            ant.SendToDongle([msg], devAntDongle, '', False, False)
+            AntDongle.Write([msg], False, False)
     
         if clv.Tacx_iVortex:
-            #---------------------------------------------------------------------------
+            #-------------------------------------------------------------------
             # Create ANT+ slave channel for VTX
-            #---------------------------------------------------------------------------
-            ant.SlaveVTX_ChannelConfig(devAntDongle, 0)
+            #-------------------------------------------------------------------
+            AntDongle.SlaveVTX_ChannelConfig(0)
 
-            #---------------------------------------------------------------------------
+            #-------------------------------------------------------------------
             # Request what DeviceID is paired to the VTX-channel
             # No pairing-loop: VTX perhaps not yet active and avoid delay
-            #---------------------------------------------------------------------------
+            #-------------------------------------------------------------------
             msg = ant.msg4D_RequestMessage(ant.channel_VTX_s, ant.msgID_ChannelID)
-            ant.SendToDongle([msg], devAntDongle, '', False, False)
+            AntDongle.Write([msg], False, False)
 
         if clv.scs != None:
-            ant.SlaveSCS_ChannelConfig(devAntDongle, clv.scs)   # Create ANT+ slave channel for SCS
-                                                  # 0: auto pair, nnn: defined SCS
+            AntDongle.SlaveSCS_ChannelConfig(clv.scs)   # Create ANT+ slave channel for SCS
+                                                        # 0: auto pair, nnn: defined SCS
             pass
     
     if not clv.gui: logfile.Console ("Ctrl-C to exit")
 
     #---------------------------------------------------------------------------
     # Loop control
     #---------------------------------------------------------------------------
-    self.RunningSwitch = True
+    # self.RunningSwitch = True             # Is set by caller!
     EventCounter       = 0
 
     #---------------------------------------------------------------------------
     # Calibrate trainer
     #---------------------------------------------------------------------------
-    Buttons         = 0
     CountDown       = 120 * 4 # 8 minutes; 120 is the max on the cadence meter
     ResistanceArray = numpy.array([0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]) # Array for running everage
     Calibrate       = 0
     StartPedalling  = True
     Counter         = 0
 
-    if devTrainer and not clv.SimulateTrainer and usbTrainer.CalibrateSupported():
+    if TacxTrainer.OK and not clv.SimulateTrainer and TacxTrainer.CalibrateSupported():

```

Comment:
Definitely possible to abstract away SimulateTrainer check by this point - CalibrateSupported() is what matters, and a simualted trainer object doesn't.

Reply:
Done